### PR TITLE
Status after wake unlock

### LINKED ISSUE #440: WDE ui does not show timer and lock after laptop wake or mfa enroll
WDE ui does not show timer and MFA lock icon in 3 scenarios . The first 2 issues were identified during the beta testing. The user has confirmed that the first 2 scenarios work intermittently. 
1. after user wake the laptop from sleep
2. after user unlock the screen.
3. user enroll mfa from WDE UI for the first time

So if WDE sends the whole status after the laptop wake/unlock, the UI can display the timer correctly.

After enrolling the MFA from WDE, UI expects the first notification from WDE, so that also has to be fixed. Currently WDE does not send the notification, if the user enrols mfa from the WDE.

### PR DESCRIPTION
**Pull request by mary-dcouto**
close #440

### CODE REVIEW

**Review by JeremyTellier — APPROVED** [review]

File: service/ziti-tunnel/service/ipc.go [hunk 1]
```diff
@@ -127,6 +127,11 @@ func SubMain(ops chan string, changes chan<- svc.Status, winEvents <-chan Window
 						}
 					}
 				}
+				rts.BroadcastEvent(dto.TunnelStatusEvent{
```
Comment (dovholuknf) [inline service/ziti-tunnel/service/ipc.go:130]: just a comment, this sends the total status to the UI - i guess i don't understand why? the UI should just get a "the service has just woken up" event and then the UI could come back to get status as needed? I would think we'd send a different event type? i just don't see the need for sending the full state to the UI at this point

File: service/cziti/sdk.go [hunk 2]
```diff
@@ -656,68 +706,25 @@ func eventCB(ztx C.ziti_context, event *C.ziti_event_t) {
...
-				return true
 
-			})
-			if noTimeoutSvc {
-				maximumTimeout = -1
-			}
-			if noTimeoutRemSvc {
-				maximumTimeoutRem = -1
-			}
+		if len(servicesToAdd) > 0 || len(servicesToRemove) > 0 {
```
Comment (dovholuknf) [inline service/cziti/sdk.go:710]: why not do this for changed services too? shouldn't this be for 'all service changes'?

File: service/cziti/sdk.go [hunk 3]
```diff
@@ -656,68 +706,25 @@ func eventCB(ztx C.ziti_context, event *C.ziti_event_t) {
 				servicesToAdd = append(servicesToAdd, svcToAdd)
 			}
 		}
-		var minimumTimeout int32 = -1
```
Comment (dovholuknf) [inline service/cziti/sdk.go:659]: all this logic moved - but i don't see UpdateMFATime() called now? is that on purpose? seems like maybe a bug? please check

Comment (dovholuknf) [reply service/cziti/sdk.go:659]: i also didn't diff the changes. i am simply assuming you jsut moved all this logic?

File: service/cziti/mfa.go [hunk 4]
```diff
@@ -80,12 +81,17 @@ func ziti_mfa_enroll_cb_go(_ C.ziti_context, status C.int, enrollment *C.ziti_mf
...
 
 // requires that the identity already be fully authenticated and used specifically for enrollment
-func VerifyMFA(id *ZIdentity, code string) {
+func VerifyMFA(id *ZIdentity, code string) error {
 	ccode := C.CString(code)
 	defer C.free(unsafe.Pointer(ccode))
 
 	log.Tracef("verifying MFA for fingerprint: %s using code: %s", id.Fingerprint, code)
 	C.ziti_mfa_verify(id.czctx, ccode, C.ziti_mfa_cb(C.ziti_mfa_cb_verify_go), unsafe.Pointer(C.CString(id.Fingerprint)))
+	authVerifyResult := strings.TrimSpace(<-mfaAuthVerifyResults)
```
Comment (dovholuknf) [inline service/cziti/mfa.go:90]: this changes the function to effectively be blocking. i assume that you've tested all the mfa scenarios? this along with the change on line 117 have me worried. as long as you feel like this logic is 'fine' and tested, then this change is intentional?

File: service/cziti/sdk.go (hunk 2 above)
Comment (mary-dcouto) [reply service/cziti/sdk.go:710]: changed services are already copied to the servicesToRemove and servicesToAdd lists

File: service/cziti/sdk.go (hunk 3 above)
Comment (mary-dcouto) [reply service/cziti/sdk.go:659]: UpdateMFATime call is in line 711 now. I copied the whole logic to the UpdateMFATime  function . I didnt change it

File: service/cziti/mfa.go (hunk 4 above)
Comment (mary-dcouto) [reply service/cziti/mfa.go:90]: I have tested it. It looks fine

File: service/ziti-tunnel/service/ipc.go (hunk 1 above)
Comment (JeremyTellier) [reply service/ziti-tunnel/service/ipc.go:130]: It looked like timers were stopping and the state was out of whack after a sleep/start. but newest tests on the the UI branch look like the timers are still going for some reason. If someone can validate that the state does not get out of whack, then I do not need the data. But from initial tests, the state gets lost on sleep/wake.

File: service/ziti-tunnel/service/ipc.go [hunk 5]
```diff
@@ -1239,6 +1248,10 @@ func broadcastNotification(adhoc bool) {
 			// do nothing
 		}
 		if len(notificationMessage) > 0 {
+			if !id.Notified {
```
[the remark this answers is not in the document]
Comment (mary-dcouto) [reply service/ziti-tunnel/service/ipc.go:1251]: added log message

File: service/ziti-tunnel/service/ipc.go (hunk 1 above)
Comment (mary-dcouto) [reply service/ziti-tunnel/service/ipc.go:130]: Suren also mentioned in his document that this issue is intermittent.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**